events/api.py

```diff
@@ -1410,6 +1413,69 @@ def get_queryset(self):
                 queryset = queryset.none()
         return queryset
 
+    @staticmethod
+    def _get_web_store_objects(organization, relation_name):
+        objects = getattr(organization, relation_name).filter(active=True)
+        if objects.exists():
+            return objects
+
+        for ancestor in organization.get_ancestors(ascending=True):
+            relation = getattr(ancestor, relation_name)
+            if relation.filter(active=True).exists():
+                return relation.filter(active=True)
+
+        return objects
```

Comment:
I think it would be a bit more robust to return a list from this function to prevent possible re-evaluation of the queryset, which can lead to an extra db hit and even (**very** theoretically) inconsistent results between the hits.

Reply:
True, and taking another look at my own code now, this can be simplified too. Implemented the suggested change + some simplifications. Also changed the endpoints return a 200/OK result with an empty list if there are no merchants or accounts.